lib/components/prometheusmetrics.js

```diff
@@ -208,6 +208,122 @@ PrometheusMetrics.prototype._register_process_metrics = function() {
     this.refresh();
 };
 
+/**
+ * Registers some exported metrics that relate to operations of the embedded
+ * matrix-js-sdk. In particular, a metric is added that counts the number of
+ * calls to client API endpoints made by the client library.
+ */
+PrometheusMetrics.prototype.registerMatrixSdkMetrics = function() {
+    var callCounts = this.addCounter({
+        name: "matrix_api_calls",
+        help: "Count of the number of Matrix client API calls made",
+    });
+
+    /*
+     * We'll now annotate a bunch of the methods in MatrixClient to keep counts
+     * of every time they're called. This seems to be neater than trying to
+     * intercept all HTTP requests and try to intuit what internal method was
+     * invoked based on the HTTP URL.
+     */
+
+    var matrixClientPrototype = require("matrix-js-sdk").MatrixClient.prototype;
+
+    var CLIENT_METHODS = [
+        "ban",
+        "createAlias",
```

Comment:
Why are we doing this? I thought the plan was to hook into the underlying request function?

I don't like this in its current form because:
 - This list will get out of date with the JS SDK as new APIs are added or new convenience functions are added.
 - This list is a PITA to maintain.
 - This list does not cover the full scope of API calls that a bridge can do (e.g. it doesn't catch raw API calls the IRC bridge does to `/initialSync`).

I am not convinced we need this. Please convince me otherwise.